parts/k8s/cloud-init/artifacts/mountetcd.sh

```diff
@@ -2,20 +2,26 @@
 # Mounting is done here instead of etcd because of bug https://bugs.launchpad.net/cloud-init/+bug/1692093
 # Once the bug is fixed, replace the below with the cloud init changes replaced in https://github.com/Azure/aks-engine/pull/661.
 set -x
-DISK=/dev/sdc
-PARTITION=${DISK}1
-MOUNTPOINT=/var/lib/etcddisk
+ETCDDISK=""
+PARTITION=""
+for DISK in $(grep -o -G "sd[a-z]" /proc/partitions | uniq); do
+  if ! grep "$DISK"1 /proc/partitions; then
+    if [[ -n $PARTITION ]]; then
+      exit 1
+    fi
+    ETCDDISK=/dev/${DISK}
+    PARTITION=${ETCDDISK}1
+  fi;
+done
 udevadm settle
+MOUNTPOINT=/var/lib/etcddisk
 mkdir -p $MOUNTPOINT
-if mount | grep $MOUNTPOINT; then
-  echo "disk is already mounted"
-  umount /dev/sdc1
-fi
-if ! grep "/dev/sdc1" /etc/fstab; then
-  echo "$PARTITION       $MOUNTPOINT       auto    defaults,nofail       0       2" >>/etc/fstab
+umount $MOUNTPOINT
```

Comment:
We can simply always run the `umount` here. If the script only has to be run once, then this will complain:

```
+ umount /var/lib/etcddisk
umount: /var/lib/etcddisk: not mounted.
```

And we just ignore that error and continue along.